frontend/src/app/utils/hooks.tsx

```diff
@@ -21,4 +21,23 @@ const useOutsideClick = (
   }, [ref, onClickOutside]);
 };
 
-export { useOutsideClick };
+const useDocumentTitle = (title: string, retainOnUnmount = false) => {
+  const defaultTitle = useRef<string>(document.title);
+
+  useEffect(() => {
+    document.title = title + " | SimpleReport";
+  }, [title]);
+
+  useEffect(() => {
+    const defaultCurrentTitle = defaultTitle.current;
+
+    return () => {
+      if (!retainOnUnmount) {
+        document.title = defaultCurrentTitle;
+      }
+    };
+    //eslint-disable-next-line
+  }, []);
```

Comment:
Can this work some way without disabling the dependency array warning? I really don't like disabling the dependency array warning, it [usually comes back to bite you later](https://github.com/facebook/create-react-app/issues/6880#issuecomment-485912528).

Reply:
I also avoid it, but in this case, I wonder that it will work on the mount and unmount component, let me try if it will work with dependencies